src/java/org/apache/cassandra/schema/SchemaKeyspace.java

```diff
@@ -955,6 +962,10 @@ private static TableMetadata fetchTable(String keyspaceName, String tableName, T
     static TableParams createTableParamsFromRow(UntypedResultSet.Row row)
     {
         return TableParams.builder()
+                          // allow_auto_snapshot column was introduced in 4.2
+                          .allowAutoSnapshot(row.has("allow_auto_snapshot") ?
+                                             row.getBoolean("allow_auto_snapshot", DatabaseDescriptor.isAutoSnapshot()) :
+                                             DatabaseDescriptor.isAutoSnapshot())
```

Comment:
Again, when it comes to schema things, `DatabaseDescriptor.isAutoSnapshot()` shouldn't be relevant. Only set the value on the builder at all if the column is present in the row. Otherwise let it pick up the default builder value of `true` and later defer to `ColumnFamilyStore.isAutoSnapshot()` to see if both the per-table setting and the node-local one are true.